fs/metrics_prometheus.cc

```diff
@@ -0,0 +1,108 @@
+#include "metrics_prometheus.h"
+
+#include <prometheus/collectable.h>
+#include <prometheus/counter.h>
+#include <prometheus/registry.h>
+
+#include <chrono>
+#include <cstdint>
+#include <memory>
+#include <utility>
+
+using namespace ROCKSDB_NAMESPACE;
+using namespace prometheus;
+
+ZenFSPrometheusMetrics::ZenFSPrometheusMetrics() {
+  registry_ = std::make_shared<Registry>();
+
+  for (auto &label_with_type : info_map_)
+    AddReporter(static_cast<uint32_t>(label_with_type.first),
+                static_cast<uint32_t>(label_with_type.second.second));
+
+  stop_collect_thread_.store(false);
+  collect_thread_ = new std::thread(&ZenFSPrometheusMetrics::run, this);
+}
+
+ZenFSPrometheusMetrics::~ZenFSPrometheusMetrics() {
+  stop_collect_thread_.store(true);
+  collect_thread_->join();
+}
+
+void ZenFSPrometheusMetrics::run() {
+  Exposer server{"127.0.0.1:8080"};
+  server.RegisterCollectable(registry_);
+
+  auto wake = std::chrono::steady_clock::now() +
+              std::chrono::milliseconds(this->report_interval_ms_);
+  while (!stop_collect_thread_.load()) {
+    std::this_thread::sleep_until(wake);
+    wake = wake + std::chrono::milliseconds(this->report_interval_ms_);
```

Comment:
It will also drift over time :shrug: By scheduling a wake up time based on the first wake up time, we eliminate drift caused by jitter in the `sleep_for` function.